src/commands/auth.ts

```diff
@@ -1,32 +1,120 @@
-import { open } from "openurl";
-import * as keytar from "keytar";
-import readline from "readline";
+import dotenv from 'dotenv';
+import open from "open";
+import readline from 'readline/promises';
+import fs from 'fs/promises';
+import path from 'path';
+import os from 'os';
+import crypto from 'crypto';
+import keytar from 'keytar';
+
+dotenv.config();
 
 const SERVICE_NAME = "AnyFlowCLI";
 const ACCOUNT_NAME = "apiToken";
+const ENCRYPTION_KEY = process.env.ENCRYPTION_KEY;
+
+if (!ENCRYPTION_KEY) {
+  console.error("ENCRYPTION_KEY is not set in the environment variables, run `anyflow init` to set it up.");
+  process.exit(1);
+}
 
 export async function authenticate() {
   console.log("Opening your browser to authenticate...");
 
-  const url = "https://anyflow.pro/api-tokens";
-  open(url, (err) => {
-    if (err) {
-      console.error(`Error opening URL: ${err}`);
-    }
-  });
+  const url = "https://app-staging.anyflow.pro/dev";
+
+  try {
+    await open(url);
+    console.log("Browser opened successfully.");
+  } catch (err) {
+    console.error(`Error opening URL: ${err}`);
+    process.exit(1);
+  }
 
   const rl = readline.createInterface({
     input: process.stdin,
     output: process.stdout,
   });
 
-  rl.question("Please paste your API token here: ", async (token) => {
-    if (token) {
-      await keytar.setPassword(SERVICE_NAME, ACCOUNT_NAME, token);
+  try {
+    const token = await rl.question("Please paste your API token here: ");
+    if (token && token.trim()) {
+      await storeToken(token.trim());
       console.log("Token stored securely.");
     } else {
-      console.log("Invalid token.");
+      console.log("Invalid token. Please provide a non-empty token.");
+      process.exit(1);
     }
+  } catch (error) {
+    console.error("Error storing token:", error);
+    process.exit(1);
+  } finally {
     rl.close();
-  });
+  }
```

Comment:
I think right after storing the token we should make an API call to test if it works. You can call `GET /api/user` to get user info and store into keytar's ACCOUNT.

Reply:
Added